Sources/Markdown/Interpretive Nodes/Aside.swift

```diff
@@ -21,87 +21,116 @@ import Foundation
-
+        public static let copyright = Kind(rawValue: "Copyright")!
+        
         /// A "date" aside.
-        case date = "Date"
-
+        public static let date = Kind(rawValue: "Date")!
+        
         /// An "invariant" aside.
-        case invariant = "Invariant"
-
+        public static let invariant = Kind(rawValue: "Invariant")!
+        
         /// A "mutatingVariant" aside.
-        case mutatingVariant = "MutatingVariant"
-
+        public static let mutatingVariant = Kind(rawValue: "MutatingVariant")!
+        
         /// A "nonMutatingVariant" aside.
-        case nonMutatingVariant = "NonMutatingVariant"
-
+        public static let nonMutatingVariant = Kind(rawValue: "NonMutatingVariant")!
+        
         /// A "postcondition" aside.
-        case postcondition = "Postcondition"
-
+        public static let postcondition = Kind(rawValue: "Postcondition")!
+        
         /// A "precondition" aside.
-        case precondition = "Precondition"
-
+        public static let precondition = Kind(rawValue: "Precondition")!
+        
         /// A "remark" aside.
-        case remark = "Remark"
-
+        public static let remark = Kind(rawValue: "Remark")!
+        
         /// A "requires" aside.
-        case requires = "Requires"
-
+        public static let requires = Kind(rawValue: "Requires")!
+        
         /// A "since" aside.
-        case since = "Since"
-
+        public static let since = Kind(rawValue: "Since")!
+        
         /// A "todo" aside.
-        case todo = "ToDo"
-
+        public static let todo = Kind(rawValue: "ToDo")!
+        
         /// A "version" aside.
-        case version = "Version"
-
+        public static let version = Kind(rawValue: "Version")!
+        
         /// A "throws" aside.
-        case `throws` = "Throws"
-
+        public static let `throws` = Kind(rawValue: "Throws")!
+        
+        /// A collection of preconfigured aside kinds.
+        public static var allCases: [Aside.Kind] {
+            [
+                note,
+                tip,
+                important,
+                experiment,
+                warning,
+                attention,
+                author,
+                authors,
+                bug,
+                complexity,
+                copyright,
+                date,
+                invariant,
+                mutatingVariant,
+                nonMutatingVariant,
+                postcondition,
+                precondition,
+                remark,
+                requires,
+                since,
+                todo,
+                version,
+                `throws`,
+            ]
+        }
+        
+        /// The underlying raw string value.
+        public var rawValue: String
+        
+        /// Creates an aside kind with the specified raw value.
+        /// - Parameter rawValue: The string the aside displays as its title.
         public init?(rawValue: String) {
```

Comment:
```suggestion
        public init(rawValue: String) {
```

Reply:
Since the initializer never fails. We should use a non-failable initializer and update all the `Aside(rawValue: xx)!` to `Aside(rawValue: xx)`

Reply:
Hey @Kyle-Ye. 👋 

The initializer needs to be failable to meet the requirements of `RawRepresentable`. By keeping it failable, it also means existing client code will continue to work instead of introducing additional breaking changes.

Right now the initializer never fails, but in the future we may want to change this. For  example, returning `nil` if `rawValue` is empty or contains only whitespace.

Reply:
+1 on keeping the type `RawRepresentable` and hence, making the initializer failable. This type is essentially a wrapper for raw string values, so `RawRepresentable` makes sense.

Reply:
> Hey @Kyle-Ye. 👋
> 
> The initializer needs to be failable to meet the requirements of `RawRepresentable`. By keeping it failable, it also means existing client code will continue to work instead of introducing additional breaking changes as we move from an `enum` to a `struct`.
> 
> Right now the initializer never fails, but in the future we may want to change this. For example, returning `nil` if `rawValue` is empty or contains only whitespace.

1. We can use a non-failable init to meet failable init requirement. That's totally OK in swift.
2. But considering the original enum's init is `init?(rawValue: String)` and we may return `nil` in the future. I agree to keep the failable init behavior.